src/Database/Adapter/Mongo/MongoClient.php

```diff
@@ -0,0 +1,563 @@
+  }
+
+  /**
+   * Find a document/s.
+   * 
+   * @param string $collection
+   * @param array $filters
+   * @param array $options
+   * 
+   * @return array
+   */
+  // https://docs.mongodb.com/manual/reference/command/find/#mongodb-dbcommand-dbcmd.find
+  public function find(string $collection, $filters = [], $options = []) {
+    
+    $result =  $this->query(array_merge([
+      MongoCommand::FIND => $collection,
+      'filter' => $this->toObject($filters),
+      ], $options)
+    );
+
+    return $result;
+  }
+
+  /**
+   * Find and modify document/s.
+   * 
+   * @param string $collection
+   * @param array $filters
+   * @param array $options
+   * 
+   * @return array
+   */
+  
+  // https://docs.mongodb.com/manual/reference/command/findAndModify/#mongodb-dbcommand-dbcmd.findAndModify
+  public function findAndModify(string $collection, $document, $update, $remove = false, $filters = [], $options = []) {
+    return $this->query(array_merge([
+      MongoCommand::FIND_AND_MODIFY => $collection,
+      'filter' => $this->toObject($filters),
+      'remove' => $remove,
+      'update' => $update,
+      ], $options)
+    );
+  }
+
+  /**
+   * Delete a document/s.
+   * 
+   * @param string $collection
+   * @param array $filters
+   * @param array $options
+   * 
+   * @return array
+   */
+  
+  // https://docs.mongodb.com/manual/reference/command/delete/#mongodb-dbcommand-dbcmd.delete
+  public function delete(string $collection, $filters = [], $limit = 1, $deleteOptions = [], $options = []) {
+    return $this->query(array_merge([
+      MongoCommand::DELETE => $collection,
+      'deletes' => [
+        $this->toObject(array_merge(
+          [
+            'q' => $this->toObject($filters),
+            'limit' => $limit,
+          ], $deleteOptions)
+        ),
+      ]],
+      $options)
+    );
+  }
+
+  /**
+   * Count documents.
+   * 
+   * @param string $collection
+   * @param array $filters
+   * @param array $options
+   * 
+   * @return array
+   */
+  
+  public function count($collection, $filters, $options) {
+    $result = $this->find($collection, $filters, $options);
+    $list = $result->cursor->firstBatch;
+    
+    return \count($list);
+  }
+
+  /**
+   * Aggregate a collection pipeline.
+   * 
+   * @param string $collection
+   * @param array $pipeline
+   
+   * 
+   * @return array
+   */
+  
+  public function aggregate($collection, $pipeline) {
+    $result = $this->query([
+      MongoCommand::AGGREGATE => $collection,
+      'pipeline' => $pipeline,
+      'cursor' => $this->toObject([]),
+    ]);
+
+    return $result;
+  }
+
+  /**
+   * Convert an assoc array to an object (stdClass).
+   * 
+   * @param array $dict
+   * 
+   * @return stdClass
+   */
+  
+  public function toObject($dict) {
+    $obj = new \stdClass();
+
+    foreach($dict as $k => $v) {
+      $key = $k == 'id' ? '_id' : $k;
+      $val = $v;
+
+      if($k == '_id') {
+        $val = new \MongoDB\BSON\ObjectId($v);
```

Comment:
```suggestion
        $val = new \MongoDB\BSON\ObjectId($val);
```

wouldn't this do the same and we can skip the temporary variable?